hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/StreamEventJournalP.java

```diff
@@ -0,0 +1,408 @@
+                                                        Function<E, T> projection,
+                                                        boolean startFromLatestSequence) {
+
+        return processorToPartitions(count, ownedPartitions)
+                .values().stream()
+                .map(partitions -> !partitions.isEmpty()
+                        ? new StreamEventJournalP<>(eventJournalReader, partitions, predicate,
+                        projection, startFromLatestSequence)
+                        : Processors.noop().get()
+                )
+                .collect(toList());
+    }
+
+    interface SerializablePredicate<E> extends com.hazelcast.util.function.Predicate<E>, Serializable {
+    }
+
+    private static class RemoteClusterMetaSupplier<E, T> implements ProcessorMetaSupplier {
+
+        static final long serialVersionUID = 1L;
+
+        private final SerializableClientConfig serializableConfig;
+        private final DistributedFunction<HazelcastInstance, EventJournalReader<E>> eventJournalReaderSupplier;
+        private final DistributedPredicate<E> predicate;
+        private final DistributedFunction<E, T> projection;
+        private final boolean startFromLatestSequence;
+
+        private transient int remotePartitionCount;
+
+        RemoteClusterMetaSupplier(
+                ClientConfig clientConfig,
+                DistributedFunction<HazelcastInstance, EventJournalReader<E>> eventJournalReaderSupplier,
+                DistributedPredicate<E> predicate,
+                DistributedFunction<E, T> projection,
+                boolean startFromLatestSequence) {
+            this.serializableConfig = new SerializableClientConfig(clientConfig);
+            this.eventJournalReaderSupplier = eventJournalReaderSupplier;
+            this.predicate = predicate;
+            this.projection = projection;
+            this.startFromLatestSequence = startFromLatestSequence;
+        }
+
+        @Override
+        public void init(@Nonnull Context context) {
+            HazelcastInstance client = newHazelcastClient(serializableConfig.asClientConfig());
+            try {
+                HazelcastClientProxy clientProxy = (HazelcastClientProxy) client;
+                remotePartitionCount = clientProxy.client.getClientPartitionService().getPartitionCount();
+            } finally {
+                client.shutdown();
+            }
+        }
+
+        @Override @Nonnull
+        public Function<Address, ProcessorSupplier> get(@Nonnull List<Address> addresses) {
+            // assign each remote partition to a member
+            Map<Address, List<Integer>> membersToPartitions =
+                    IntStream.range(0, remotePartitionCount)
+                             .boxed()
+                             .collect(groupingBy(partition -> addresses.get(partition % addresses.size())));
+
+            return address -> new RemoteClusterProcessorSupplier<>(membersToPartitions.get(address),
```

Comment:
That's what I thought - but can we distribute the partitions then according to the partition table or do we not care about that?